docs/src/modules.md

```diff
@@ -34,44 +34,47 @@ Other files needed by the module can also be stored within the
-This is similar to
-[`mkEnableOption`](https://nix-community.github.io/docnix/reference/lib/options/lib-options-mkenableoption/)
-from the standard library, however it integrates with
-[`stylix.enable`](./options/nixos.md#stylixenable) and
-[`stylix.autoEnable`](./options/nixos.md#stylixautoenable)
-and generates more specific documentation.
+All modules should be created using `mkTarget`.
 
 A general format for modules is shown below.
 
 ```nix
+mkTarget:
 { config, lib, ... }:
+mkTarget
 {
-  options.stylix.targets.«name».enable =
-    config.lib.stylix.mkEnableTarget "«human readable name»" true;
+  name = "«target attribute name»";
+  humanName = "«human readable name»";
 
-  config =
-    lib.mkIf (config.stylix.enable && config.stylix.targets.«name».enable)
+  configElements = [
+    { colors }:
       {
         programs.«name».backgroundColor = config.lib.stylix.colors.base00;
       };
+  ];
 }
 ```
 
-The human readable name will be inserted into the following sentence:
+### `mkTarget`
 
-> Whether to enable theming for «human readable name».
+The `mkTarget` function is a special function which automatically generates
+options and guards configuration with`stylix.${attr} != null` on configuration
+to cut down on boilerplate.
 
-If your module will touch options outside of `programs.«name»` or `services.«name»`,
-it should include an additional condition in `mkIf` to prevent any effects
-when the target is not installed.
+Arguments without defaults must be set explicitly.
 
-The boolean value after `mkEnableTarget` should be changed to `false` if
-one of the following applies:
+| Argument | Type | Default | Description | Example |
+|----------|------|---------|-------------|---------|
+|`name`|String||Used when generating options; `stylix.targets.${name}`|"firefox"|
+|`humanName`|String||Used when generating enable option; `stylix.targets.${name}.enable` has a description of "Whether to enable theming for ${humanName}."|"Firefox"|
+|`autoEnableConditions`|Boolean|true|Used for determining when an attribute called `autoEnable` should be set to `false` if one of the following applies:<ul><li>The module requires further manual setup to work correctly.</li><li>There is no reliable way to detect whether the target is installed, *and* enabling it unconditionally would cause problems.</li></ul>|`pkgs.stdenv.hostPlatform.isLinux`|
+|`extraOptions`|Attrset of options|{ }|Attributes will be mapped to `stylix.targets.${name}.${option}`|`{ colorTheme.enable = lib.mkEnableOption "color theme"; }`|
+|`configElements`|List of function returning attrset|{ }|Each function should take the Stylix attributes that it needs (e.g. `colors`, `image`, `fonts`) and `cfg` if necessary. This `cfg` is an alias for `config.stylix.targets.${name}`. Each function should return an attribute set to be merged into `config`. If any of the Stylix attributes needed by a function are `null`, the configuration will not be merged.|*See above template*|
```

Comment:
Nice to haves:

It being a list could be optional; we could normalise internally with `lib.toList`

Elememts being a function could be optional; we could check using `isFunction` and/or `isAttrs`.